### DISCUSSION TITLE
Host provider auto-detection regarding GitLab

### DISCUSSION THREAD

**Discussion by HarryHartmann** [git-ecosystem/git-credential-manager#1433] [Q&A]
In my eco-environment I am trying to apply standard git command to add a submodule:

`git submodule add <url_to_git_repo> <local_dir_path>`

The repo _url_to_git_repo_ is a GitLab hosted repository. The "git submodule add" command execution responds with error message: 

warning: failed to probe 'URL2REPO' to detect provider
warning: The ServicePointManager does not support proxies with the URL2REPO scheme.
warning: see https://aka.ms/gcm/autodetect for more information.
fatal: The ServicePointManager does not support proxies with the URL2REPO scheme.

"Git Credential Manager (GCM) supports authentication with multiple different Git host providers including: GitHub, Bitbucket, and Azure Repos." I don't see GitLab listed as supported Git host.

What do I miss to get it working? Any hint helps, thnx, Harry.

**Comment by hickford** [comment]
Have you tried manual configuration? https://github.com/git-ecosystem/git-credential-manager/blob/main/docs/autodetect.md#manual-configuration 

```sh
git config --global credential.gitlab.example.com.provider gitlab
```

**Comment by ldennington** [comment]
> "Git Credential Manager (GCM) supports authentication with multiple different Git host providers including: GitHub, Bitbucket, and Azure Repos." I don't see GitLab listed as supported Git host.

@HarryHartmann - to clarify, even though it's not listed, GCM supports authentication with GitLab. Were you able to try hickford's suggestion of manual configuration to see if that resolves your issue?
